### PR TITLE
Assignment18_Harshal

### PR DESCRIPTION
**Pull request by harshalrane27**
Assignment18_Harshal

1) Write a program to rearrange String in given format. 
String str = "te2cH4no";
output : "24tecHno";
Hint : AllDigitAllCharacter

image
-------------------------------------------
2) Write a program to rearrange String in given format. 
Hint : Uppercase letters + digits + LowerCase letter
input = "t8EcHNo5Cred3iTs";
output : EHNCT853tcoredis 

image
---------------------------------------------
3) Find frequency of given character from given String. 
String str = "TechnoTtcredits";
char targetChar = 't';
output : 4

image

### CODE REVIEW

File: src/harshalRane/Assignments/assignment18_rearrangeFrequency/Assignment18FindFrequencyOfChar.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+public class Assignment18FindFrequencyOfChar {
+	static int FindFrequencyOfLetter(String input, char targetChar) {
+		int count = 0;
+		input = input.toLowerCase();
+		targetChar = Character.toLowerCase(targetChar);
+
+		for (int index = 0; index < input.length(); index++) {
+			char ch = input.charAt(index);
+			if (ch == 't') {
```
Comment (KrishnaKTechnocredits) [inline src/harshalRane/Assignments/assignment18_rearrangeFrequency/Assignment18FindFrequencyOfChar.java:11]: why 't', it should be targetChar.

File: src/harshalRane/Assignments/assignment18_rearrangeFrequency/Assignment18RearrangeAndFindFrequency.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		}
+
+		static int FindFrequencyOfLetter(String input, char targetChar) {
+			int count = 0;
+			input = input.toLowerCase();
+			targetChar = Character.toLowerCase(targetChar);
+
+			for (int index = 0; index < input.length(); index++) {
+				char ch = input.charAt(index);
+				if (ch == 't') {
```
Comment (KrishnaKTechnocredits) [inline src/harshalRane/Assignments/assignment18_rearrangeFrequency/Assignment18RearrangeAndFindFrequency.java:42]: we should not hardcode 't' here

**Review by KrishnaKTechnocredits — CHANGES_REQUESTED** [review]
Harshal, please update code as requested

File: src/harshalRane/Assignments/assignment18_rearrangeFrequency/Assignment18FindFrequencyOfChar.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package harshalRane.Assignments.assignment18_rearrangeFrequency;
+
+public class Assignment18FindFrequencyOfChar {
+	static int FindFrequencyOfLetter(String input, char targetChar) {
```
Comment (harshvegada) [inline src/harshalRane/Assignments/assignment18_rearrangeFrequency/Assignment18FindFrequencyOfChar.java:4]: method name should be in small

File: src/harshalRane/Assignments/assignment18_rearrangeFrequency/Assignment18RearrangeAndFindFrequency.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+				if (Character.isDigit(ch)) {
+					digit = digit + ch;
+				} else if (Character.isLetter(ch)) {
+					letter = letter + ch;
+				}
+			}
+			return digit + letter;
+		}
+
+		static String RearrangeLetterDigit(String name) {
```
Comment (harshvegada) [inline src/harshalRane/Assignments/assignment18_rearrangeFrequency/Assignment18RearrangeAndFindFrequency.java:18]: same as above

**Review by harshvegada — APPROVED** [review]
Code looks good to me, Approved

File: src/harshalRane/Assignments/assignment18_rearrangeFrequency/Assignment18FindFrequencyOfChar.java (hunk 3 above)
Comment (harshalrane27) [reply src/harshalRane/Assignments/assignment18_rearrangeFrequency/Assignment18FindFrequencyOfChar.java:4]: yes.. will take care next time

**Pull request merged**